src/java/org/apache/cassandra/tools/WapModifyAndGenerateSSTablesUtility.java

```diff
@@ -0,0 +1,937 @@
+     *
+     * @param String path of dirtectory
+     * @return list of abssolute path of all the sstables
+     */
+    private static List<String> generateAbsolutePathOfAllSSTables(String directoryPath) {
+	List<String> absolutePathOfAllSSTables = new ArrayList<String>();
+	File dir = new File(directoryPath);
+	if (dir.isDirectory()) {
+	    File[] directoryListing = dir.listFiles();
+	    for (File file : directoryListing) {
+		if (file.getName().endsWith("Data.db")) {
+		    absolutePathOfAllSSTables.add(file.getAbsolutePath());
+		}
+	    }
+	    return absolutePathOfAllSSTables;
+	}
+	absolutePathOfAllSSTables.add(directoryPath);
+	return absolutePathOfAllSSTables;
+    }
+
+    /**
+     * To generate the directory
+     *
+     * @param String path to directory
+     */
+    private static File createDirectory(String pathToDirectory) {
+	File modifiedDataDirectory = new File(pathToDirectory);
+
+	if (!modifiedDataDirectory.isDirectory()) {
+	    if (modifiedDataDirectory.mkdirs()) {
+	    } else {
+		System.err.println("unable to create direcotry for new generated sstables.");
+		throw new IllegalArgumentException("don't have permission to create directory");
+	    }
+	}
+	return modifiedDataDirectory;
+    }
+
+    /**
+     * To generate file
+     *
+     * @param String path of file
+     */
+    private static File createFile(String filePath) {
+	File file = new File(filePath);
+	try {
+	    file.createNewFile();
+	} catch (IOException e) {
+	    System.err.println("Unable to create log file");
+	    e.printStackTrace();
+	}
+	return file;
+    }
+
+    /**
+     * @purpose To copy single file
+     * @param source file
+     * @param destfile
+     */
+    @SuppressWarnings("resource")
+    public static void copyFile(File from, File to) throws IOException {
+
+	if (!to.exists()) {
+	    to.createNewFile();
+	}
+
+	try (FileChannel in = new FileInputStream(from).getChannel();
+		FileChannel out = new FileOutputStream(to).getChannel()) {
+
+	    out.transferFrom(in, 0, in.size());
+	}
+    }
+
+    /**
+     *
+     * @purpose To copy whole directory
+     * @param source Directory
+     * @param dest.Directory
+     *
+     */
+    public static void copyDirectory(File from, File to) throws IOException {
+
+	if (!to.exists()) {
+	    to.createNewFile();
+	}
+
+	if (from.isDirectory()) {
+	    String[] children = from.list();
+	    for (String child : children) {
+		copyDirectory(new File(from, child), new File(to, child));
+	    }
+	} else {
+	    copyFile(from, to);
+	}
+    }
+
+    /**
+     * @purpose : To remove all files in given directory
+     * @param File Name
+     */
+    public static void cleanDirectory(File file) {
+	if (!file.isDirectory()) {
+	    System.out.println("File does not exist!!");
+	    return;
+	}
+	for (File child : file.listFiles()) {
+	    child.delete();
+	}
+    }
+
+    /**
+     * @purpose To display at both level
+     * @param log file
```

Comment:
This file contains many similar problems, please check [official document](http://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDHJECF) to learn how we can use `@param` tag